### PR TITLE
Disabling sync on android now disables all synchronizations

### PR DESCRIPTION
**Pull request by jonathanmos**
Disabling sync on android only partially disabled synchronization. In this PR all synchronizations are disabled. 

Relates to issue: #1513

### CODE REVIEW

File: detox/android/detox/src/full/java/com/wix/detox/reactnative/ReactNativeExtension.kt [hunk 1]
```diff
@@ -62,6 +62,15 @@ object ReactNativeExtension {
         }
     }
 
+    @JvmStatic
+    fun enableAllSynchronization(applicationContext: ReactApplication) {
+        val reactContext = getCurrentReactContextSafe(applicationContext)
+
+        if (reactContext != null) {
```
Comment (noomorph) [inline detox/android/detox/src/full/java/com/wix/detox/reactnative/ReactNativeExtension.kt:69]: @d4vidi , this just made me wonder – do we still not support properly non-React Android apps (e.g. using network idling resources regardless of whether it is React or not)?

**Review by noomorph — APPROVED** [review]
Provides great value while being terse. Amazing!

File: detox/android/detox/src/full/java/com/wix/detox/reactnative/ReactNativeExtension.kt (hunk 1 above)
Comment (d4vidi) [reply detox/android/detox/src/full/java/com/wix/detox/reactnative/ReactNativeExtension.kt:69]: Afraid so: #2543

File: detox/test/e2e/12.animations.test.js [hunk 2]
```diff
@@ -30,31 +28,40 @@ describe('React-Native Animations', () => {
     await element(by.id('UniqueId_AnimationsScreen_startButton')).tap();
   }
 
-  _.forEach(['JS', 'Native'], (driver) => {
-    it(`should find element (driver: ${driver})`, async () => {
+  describe.each(['JS', 'Native'])('(driver: %s)', (driver) => {
```
Comment (d4vidi) [inline detox/test/e2e/12.animations.test.js:31]: 🚀

File: detox/test/e2e/12.animations.test.js [hunk 3]
```diff
@@ -30,31 +28,40 @@ describe('React-Native Animations', () => {
...
-    it(`should wait during delays shorter than 1.5s (driver: ${driver})`, async () => {
+    it(`should wait during delays shorter than 1.5s`, async () => {
       await _startTest(driver, {delay: 500});
       await expect(element(by.id('UniqueId_AnimationsScreen_afterAnimationText'))).toExist();
     });
+
+    it(`should not wait for an animation to complete while the synchronization is disabled`, async () => {
+      await device.disableSynchronization();
+      await _startTest(driver, {duration: 5000});
+      await expect(element(by.id('UniqueId_AnimationsScreen_afterAnimationText'))).not.toExist();
```
Comment (d4vidi) [inline detox/test/e2e/12.animations.test.js:60]: This ID duplication, alongside overall matchers duplications - screams a test driver is required. Boyscouting, much?

**Review by d4vidi — APPROVED** [review]

**Pull request merged**